integr8ly/integreatly-operator
# MGDAPI-4543 - RHOAM e2e tests for GCP

# Issue link
https://issues.redhat.com/browse/MGDAPI-4543
RHOAM e2e tests for GCP installations
**Current PR require https://github.com/integr8ly/integreatly-operator/pull/3052**  (please see verification notes)

# What
E2E tests created/updated for GCP installations

The difference between AWS and GCP was taken into consideration
- Amazon RDS vs Google Cloud SQL
- Amazon ElastiCache for Redis vs Google Cloud Memorystore for Redis
- AWS S3 vs Google Cloud Storage
- NooBaa / Multi-Cloud Object Gateway (MCG) was added to RHOAM for GCP platform

Tests changes included:
- Common tests update - GCP specific conditions were added to common tests
- Functional AWS tests Refactoring - separate/extract functions that common for both AWS and GCP
- New functional tests for GCP

TESTS that were changed:
- A03, J03, A06, A09, A10, A11, A12, A13, A14, A15, A25, C01, C04, F03, F04, "Verify Alerts are not firing "  
New functional tests for GCP:
- GCP01, GCP02, GCP03, GCP04

Validation that was done:
- Following tests were executed on GCP cluster:
  A03, J03, A06, A09, A10, A11, A12, A13, A14, A15, A25, C01, C04, F03, F04, "Verify Alerts are not firing ", GCP01, GCP02, GCP03, GCP04.
- GCP tests are still in verification / waiting for CRO GetGCPProject() / infrastructure CR fix/investigation)*

Follow on task will be created for more E2E GCP tests - https://issues.redhat.com/browse/MGDAPI-5075:
  ./scripts/backup-restore/postgres.sh # => aws_postgres.sh + gcp_postgres.sh
  ./scripts/backup-restore/j05-verify-3scale-postgres-backup-and-restore.sh - check if it could use it for both AWS and GCP, by calling corresponding postgres.sh
  ./scripts/backup-restore/j07-verify-rhsso-backup-and-restore.sh - notes similar as j05
  ./scripts/backup-restore/j08-verify-user-sso-backup-and-restore.sh - notes similar as j05
- Noobaa / Multi-Cloud Object Gateway (MCG)
  check Noobaa CR
  check Noobaa backingstore

# Verification steps
I will be happy to pair with reviewer on verification, but here are the steps to do it yourself
**Notes** Please note that Current PR require  `PR 3052 (task MGDAPI-4907)` , that not merged yet.  **To make Testing easier** - the  testing branch **valery_MGDAPI-4543_4907_test** and temporary PR **https://github.com/integr8ly/integreatly-operator/pull/3073**  were created that include current PR with changes from PR 3052. These are testing steps:
1) create local branch pointed to `valery_MGDAPI-4543_4907_test`

```sh
git remote add valery git@github.com:valerymo/integreatly-operator.git
git fetch valery
git switch -c  <prefix>_MGDAPI-4543_4907_test  valery/valery_MGDAPI-4543_4907_test

$ go mod tidy
$ make vendor/fix 

$make code/compile
```
2) Create GCP cluster and install RHOAM , as noted in PR https://github.com/integr8ly/integreatly-operator/pull/3052
3) Create IDP: 
```sh
$ PASSWORD=Password1 DEDICATED_ADMIN_PASSWORD=Password1 scripts/setup-sso-idp.sh
```
3) Run following tests on GCP cluster: A03, J03, A06, A09, A10, A11, A12, A13, A14, A15, A25, C01, C04, F03, F04, "Verify Alerts are not firing " ,  GCP01, GCP02, GCP03, GCP04.
**These are sample commands for running single test(s):**  
```
$ TEST=A03 INSTALLATION_TYPE=managed-api make test/e2e/single  LOCAL=false
$ TEST="Verify Alerts are not firing" INSTALLATION_TYPE=managed-api make test/e2e/single  LOCAL=false
```
For functional tests:
```
$ TEST=F03 INSTALLATION_TYPE=managed-api make test/e2e/single  LOCAL=false BYPASS_STORAGE_TYPE_CHECK=true
$ TEST=GCP01 INSTALLATION_TYPE=managed-api make test/e2e/single  LOCAL=false BYPASS_STORAGE_TYPE_CHECK=true
```
